django/core/servers/basehttp.py

```diff
@@ -147,6 +148,19 @@ def close(self):
         self.get_stdin().read()
         super().close()
 
+    def finish_response(self):
+        try:
+            if self.environ["REQUEST_METHOD"] == "HEAD":
+                deque(self.result, maxlen=0)  # consume iterator quickly.
+                self.finish_content()
+            else:
+                if not self.result_is_file() or not self.sendfile():
+                    for data in self.result:
+                        self.write(data)
+                    self.finish_content()
+        finally:
+            self.close()
```

Comment:
I think we can use `BaseHandler.finish_response()` here by calling `super()`:
 ```suggestion
    def finish_response(self):
        if self.environ["REQUEST_METHOD"] == "HEAD":
            try:
                deque(self.result, maxlen=0)  # consume iterator quickly.
                self.finish_content()
            finally:
                self.close()
        else:
            super().finish_response()
```

Reply:
Nice :+1: updated [737df0c](https://github.com/django/django/pull/16502/commits/737df0c704f7bf051ba00e99ea847ef1032620ce)